packages/core/src/StateNode.ts

```diff
@@ -1800,18 +1800,15 @@ class StateNode<
       target,
       source: this,
       internal,
-      eventType: transitionConfig.event
-    };
-
-    Object.defineProperty(transition, 'toJSON', {
-      value: () => ({
+      eventType: transitionConfig.event,
+      toJSON: () => ({
```

Comment:
Hm, could you elaborate on what was the issue on the lower level? how changing the enumerability of this property has fixed it?

Reply:
Because of this:

```js
return toArray(resolvedTransition).map((transition) => ({
  ...transition,
  event: eventType,
  delay: resolvedDelay
}));
```

The `...transition` wasn't taking the `toJSON` property.